src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -24,29 +24,9 @@ internal interface IDiagnosticAnalyzerService
     void RequestDiagnosticRefresh();
 
     /// <summary>
-    /// Get diagnostics currently stored in the source. returned diagnostic might be out-of-date if solution has changed but analyzer hasn't run for the new solution.
+    /// Force analyzes the given project by running all applicable analyzers on the project.
     /// </summary>
-    /// <param name="workspace">Workspace for the document/project/solution to compute diagnostics for.</param>
-    /// <param name="projectId">Required project to scope the returned diagnostics.</param>
-    /// <param name="documentId">Optional document to scope the returned diagnostics. When provided, only local
-    /// diagnostics to that document are returned and non-local diagnostics are not returned. When absent, only
-    /// non-local diagnostics are included and local diagnostics are not returned.</param>
-    /// <param name="cancellationToken">Cancellation token.</param>
-    /// <remarks>
-    /// Local diagnostics are the ones that are reported by analyzers on the same file for which the callback was received
-    /// and hence can be computed by analyzing a single file in isolation.
-    /// <para/>
-    /// Non-local diagnostics are the ones reported by analyzers either at compilation end callback OR
-    /// in a different file from which the callback was made. Entire project must be analyzed to get the
-    /// complete set of non-local document diagnostics.
-    /// </remarks>
-    Task<ImmutableArray<DiagnosticData>> GetCachedDiagnosticsAsync(
-        Workspace workspace, ProjectId projectId, DocumentId? documentId, CancellationToken cancellationToken);
-
-    /// <summary>
-    /// Force analyzes the given project by running all applicable analyzers on the project and caching the reported analyzer diagnostics.
-    /// </summary>
-    Task ForceAnalyzeProjectAsync(Project project, CancellationToken cancellationToken);
+    Task<ImmutableArray<DiagnosticData>> ForceAnalyzeProjectAsync(Project project, CancellationToken cancellationToken);
```

Comment:
note: the internal api for this *already* returned these values (which is what unit tests were validating *to begin with*).  This removes any sort of impedence mismatch where tests were validating one thing, but the product code could potentially experience something else because it went through a different *untested* api.